monkey/infection_monkey/monkey.py

```diff
@@ -371,7 +372,8 @@ def cleanup(self):
             self._singleton.unlock()
         except Exception as e:
             logger.error(f"An error occurred while cleaning up the monkey agent: {e}")
-            InfectionMonkey._self_delete()
+            if deleted is None:
+                InfectionMonkey._self_delete()
```

Comment:
Deleted may be undefined, resulting in a `NameError`

Reply:
Can't, It's defined in line 346